app/keepers/keepers.go

```diff
@@ -443,12 +456,15 @@ func NewAppKeeper(
 		pfmrouterkeeper.DefaultRefundTransferPacketTimeoutTimestamp,
 	)
 	transferStack = ratelimit.NewIBCMiddleware(appKeepers.RatelimitKeeper, transferStack)
+	transferStack = ibcfee.NewIBCMiddleware(transferStack, appKeepers.IBCFeeKeeper)
 
 	// Create ICAHost Stack
 	var icaHostStack porttypes.IBCModule = icahost.NewIBCModule(appKeepers.ICAHostKeeper)
+	icaHostStack = ibcfee.NewIBCMiddleware(icaHostStack, appKeepers.IBCFeeKeeper)
 
 	// Create Interchain Accounts Controller Stack
 	var icaControllerStack porttypes.IBCModule = icacontroller.NewIBCMiddleware(nil, appKeepers.ICAControllerKeeper)
+	icaControllerStack = ibcfee.NewIBCMiddleware(icaControllerStack, appKeepers.IBCFeeKeeper)
 
 	// Create IBC Router & seal
 	ibcRouter := porttypes.NewRouter().
```

Comment:
Don't we want to add the ibcfee to the provider stack also?

Reply:
How would you incentivize a protocol generated packet?

You could perhaps do that to allow for async packet incentivization:
* https://ibc.cosmos.network/main/middleware/ics29-fee/msgs#msgpaypacketfeeasync

I think that work is outside the scope of this PR.

Reply:
With funds from the ICS rewards :)

Reply:
I agree it could be done that way. Code changes are required on ICS to make it aware of the ICS29 features.

As this is not yet done, we cannot address it at this time.

I'll opena a feature request on ICS.

Reply:
Issue opened on ICS:
* https://github.com/cosmos/interchain-security/issues/1782